depot/app/models/product.rb

```diff
@@ -1,22 +1,38 @@
 class Product < ApplicationRecord
+  # constants
+  PERMALINK_REGEX = /^[a-z0-9]+(?:-[a-z0-9]+){2,}/i
+  DESCRIPTION_REGEX = /^\s*\S+(?:\s+\S+){4,9}\s*$/
+
+
   has_many :line_items
   has_many :orders, through: :line_items
   before_destroy :ensure_not_referenced_by_any_line_item
 
 
-  validates :title, :description, :image_url, presence: true
-  validates :price, numericality: {greater_than_or_equal_to: 0.01}
+  validates :title, :description, :image_url, :permalink, presence: true
+  validates :permalink, uniqueness: true
+  validates_format_of :permalink, with: PERMALINK_REGEX, multiline: true,
```

Comment:
why are you writing different lines to validate permalink. Merge them in one validation.